grammar/it/src/rules.rs

```diff
@@ -0,0 +1,513 @@
+                              "cent" => 100,
+                              "cento" => 100,
+                              "duecento" => 200,
+                              "trecento" => 300,
+                              "quattrocento" => 400,
+                              "cinquecento" => 500,
+                              "seicento" => 600,
+                              "settecento" => 700,
+                              "ottocento" => 800,
+                              "novecento" => 900,
+                              "mil" => 1000,
+                              _ => return Err(RuleError::Invalid.into())
+                          };
+                          IntegerValue::new(value)
+                      });
+    b.rule_2("numbers 100..199",
+             integer_check_by_range!(100, 100),
+             integer_check_by_range!(0, 99),
+             |_, b| IntegerValue::new(b.value().value + 100));
+    b.rule_3("numbers 200..999",
+             integer_check_by_range!(2, 9),
+             integer_check_by_range!(100, 100),
+             integer_check_by_range!(0, 99),
+             |a, b, c| IntegerValue::new(a.value().value * b.value().value + c.value().value));
+    b.rule_1_terminal("integer (numeric)",
+                      b.reg(r#"(\d{1,18})"#)?,
+                      |text_match| IntegerValue::new(text_match.group(0).parse()?));
+    b.rule_1_terminal("integer with thousands separator .",
+                      b.reg(r#"(\d{1,3}(\.\d\d\d){1,5})"#)?,
+                      |text_match| {
+                          let reformatted_string = text_match.group(1).replace(".", "");
+                          let value: i64 = reformatted_string.parse()?;
+                          IntegerValue::new(value)
+                      });
+    b.rule_1_terminal("decimal number",
+                      b.reg(r#"(\d*,\d+)"#)?,
+                      |text_match| {
+                          let reformatted_string = text_match.group(1).replace(",", ".");
+                          let value: f32 = reformatted_string.parse()?;
+                          FloatValue::new(value)
+                      });
+    b.rule_3("number dot number",
+             number_check!(|number: &NumberValue| !number.prefixed()),
+             b.reg(r#"punto"#)?,
+             number_check!(|number: &NumberValue| !number.suffixed()),
+             |a, _, b| {
+                 Ok(FloatValue {
+                     value: b.value().value() * 0.1 + a.value().value(),
+                     ..FloatValue::default()
+                 })
+             });
+    b.rule_1_terminal("decimal with thousands separator",
+                      b.reg(r#"(\d+(\.\d\d\d)+,\d+)"#)?,
+                      |text_match| {
+                          let reformatted_string = text_match.group(1).replace(".", "").replace(",", ".");
+                          let value: f32 = reformatted_string.parse()?;
+                          FloatValue::new(value)
```

Comment:
`parse()?` will return a global error in the parsing for Japanese numbers. I am wondering if a mapping to Invalid Error is a better approach. WDYT?

In others languages, we use this approach too but we may want to raise an Invalid error instead. 

Indeed when a production rule raises an Invalid error, it lets other rules work. Others kind of errors will stop the parsing.

Reply:
Sounds good. Let's put that on the list for V1 and adjust other languages too then?